wave_kinematics/tests/rotation_test.cpp

```diff
@@ -0,0 +1,185 @@
+#include <iostream>
+#include <random>
+
+#include <gtest/gtest.h>
+
+#include "wave/kinematics/numerical_test_functors.hpp"
+#include "wave/kinematics/rotation.hpp"
+#include "wave/utils/math.hpp"
+#include "wave/utils/utils.hpp"
+
+
+class RotationTestFixture : public ::testing::Test {
+public:
+  // Data members used for testing.
+  wave::Rotation rotation1;
+  wave::Rotation rotation1_truth;
+  wave::Rotation rotation2;
+  wave::Rotation rotation3;
+  wave::Rotation rotation1_inverse;
+  wave::Rotation eye3;
+
+  // A vector v1.
+  wave::Vec3 v1;
+  // v1 rotated by rotation1.
+  wave::Vec3 v1_rotated;
+  // True angle axis params for rotation1.
+  wave::Vec3 rotation1_angleaxis_params;
+
+  void SetUp() {
+    wave::Rotation::fromEulerXYZ(0.1, 0.2, 0.3, rotation1);
+    wave::Rotation::fromEulerXYZ(-0.1, -0.2, -0.3, rotation2);
+    // rotation3 is the composition of rotation1 and rotation2.
+    wave::Rotation::fromEulerXYZ(
+      -0.065223037770317, 0.020616131403543, -0.013176146513587, rotation3);
+    // rotation1_inverse is the inverse of rotation1.
+    wave::Rotation::fromEulerXYZ(-0.037879880513201,
+                                 -0.220124031212965,
+                                 -0.285771700628461,
+                                 rotation1_inverse);
+    eye3.rotation_object.setMatrix(1, 0, 0, 0, 1, 0, 0, 0, 1);
+
+    // rotation1_true is set using the underlying element wise initializer.
+    // It is the expected rotation matrix, using the parameters from
+    // rotation1.
+
+    rotation1_truth.rotation_object.setMatrix(0.936293363584199,
+                                              -0.275095847318244,
+                                              0.218350663146334,
+                                              0.289629477625516,
+                                              0.956425085849232,
+                                              -0.036957013524625,
+                                              -0.198669330795061,
+                                              0.097843395007256,
+                                              0.975170327201816);
+
+
+    v1 << 1, 1, 1;
+    v1_rotated << 0.879548179412290, 1.209097549950123, 0.874344391414010;
+    rotation1_angleaxis_params << 0.068924613882066, 0.213225926957886,
+      0.288748939228675;
+  }
+};
+
+
+// Construction tests.
+
+// Test that fromEulerXYZ returns the correct matrix.
+TEST_F(RotationTestFixture, test_from_euler_xyz) {
```

Comment:
Having another look at the Gtest doc, it says: "Both names must be valid C++ identifiers, and they should not contain underscore (_).", this is not a "must not" but rather a "should", that said I believe we should follow the guidelines as much as possible. ([source](https://github.com/google/googletest/blob/master/googletest/docs/Primer.md#simple-tests))